common/src/main/java/io/smallrye/config/common/utils/StringUtil.java

```diff
@@ -88,14 +88,62 @@ public static String[] split(String text) {
         return list.toArray(NO_STRINGS);
     }
 
+    private static boolean isAsciiLetterOrDigit(char c) {
+        return 'a' <= c && c <= 'z' ||
+                'A' <= c && c <= 'Z' ||
```

Comment:
```suggestion
               isAsciiUpperCase(c) ||
```
Or would a method call here make things too slow? Though, it might get inlined, no?

Reply:
I am in favour of reusing what we got; in theory the size (in term of bytecode size) of both methods is so small that will likely being inlined, if hot, unless we are unlucky and the depth of the call fall on the max inlining depth (which is 15 from JDK>=11 IIRC).
In theory these methods will be called for each character of each property names, which means they will likely (to be verified, really!) C2 compiled, and inlining, as intrinsics, won't kick in with previous compilation levels AFAK, so, If lucky, everything will work as expected.
To play safe, we should manually inline everything if this is supposed to happen at startup and we won't expect C2 compilation to kick-in.

TBH in a later iteration of this code, due to https://github.com/smallrye/smallrye-config/pull/984#issuecomment-1694765001 I will likely turn this code into some branchless variant, which can work on 4 chars at time, which will give 3 important benefits:
- most comparisons are on common prefixes, hence batching will grant to speedup moving to the different part of strings to compare
- most names have a mix of alphabetical and not chars, very unpredictable; a branchless approach will pay less in such cases 
- the batchy and branchless variant will amortize the cost of unlucky inlining decisions or lower compilation level than C2 by performing more ops in a row